src/vehicle.cpp

```diff
@@ -958,31 +959,31 @@ void vehicle::do_autodrive()
         return;
     }
 
-    point omt_diff = omt_path.back().xy() - veh_omt_pos.xy();
-    if( omt_diff.x > 3 || omt_diff.x < -3 || omt_diff.y > 3 || omt_diff.y < -3 ) {
+    point_rel_omt omt_diff = omt_path.back().xy() - veh_omt_pos.xy();
+    if( omt_diff.x() > 3 || omt_diff.x() < -3 || omt_diff.y() > 3 || omt_diff.y() < -3 ) {
```

Comment:
Note to self: we also need a standard utility function for `within_box()` or something similar.

Reply:
We have `box` and `rectangle` with `contains` methods.  We just need a helper function to make a `rectangle` from a centre point and bounds.